WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -779,21 +779,32 @@ public void onUploadSuccess(MediaModel media) {
             mEditorMediaUploadListener.onMediaUploadSucceeded(String.valueOf(media.getId()),
                     FluxCUtils.fromMediaModel(media));
         }
+        removeMediaFromPendingList(media);
     }
 
     @Override
     public void onUploadCanceled(MediaModel media) {
+        removeMediaFromPendingList(media);
     }
 
     @Override
     public void onUploadError(MediaModel media, MediaStore.MediaError error) {
         AnalyticsTracker.track(Stat.EDITOR_UPLOAD_MEDIA_FAILED);
         String localMediaId = String.valueOf(media.getId());
-        if (error.type == MediaErrorType.GENERIC_ERROR) {
-            mEditorMediaUploadListener.onMediaUploadFailed(localMediaId, getString(R.string.tap_to_try_again));
-        } else {
-            mEditorMediaUploadListener.onMediaUploadFailed(localMediaId, error.message);
+
+        // Display custom error depending on error type
+        String errorMessage;
+        switch (error.type) {
+            case UNAUTHORIZED:
```

Comment:
I just tried, it looks good, we should remove the "retry" button in this case, but we already have another issue for this.